src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -833,6 +845,9 @@ def restore_key(cmd,
 
         need_change = __print_restore_preview(restore_diff, yes=yes)
 
+        if dry_run and not yes:
```

Comment:
We can  validate that dry-run and yes are not specified together and return if only dry-run is `True`